syft/frameworks/torch/tensors/interpreters/replicated_shared.py

```diff
@@ -0,0 +1,154 @@
+import random
+from operator import add, sub
+import torch
+import syft
+from syft.generic.abstract.tensor import AbstractTensor
+
+
+class ReplicatedSharingTensor(AbstractTensor):
+    def __init__(
+        self, shares_map=None, owner=None, id=None, tags=None, description=None,
+    ):
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.child = shares_map
+        self.ring_size = 2 ** 5
+
+    def share_secret(self, secret, workers):
+        number_of_shares = len(workers)
+        workers = self.__arrange_workers(list(workers))
+        shares = self.generate_shares(secret, number_of_shares)
+        shares_map = self.__distribute_shares(workers, shares)
+        self.child = shares_map
+        return self
+
+    @staticmethod
+    def __arrange_workers(workers):
+        """ having local worker in index 0 saves one communication round"""
+        if syft.hook.local_worker in workers:
+            workers.remove(syft.hook.local_worker)
+            workers = [syft.hook.local_worker] + workers
+        return workers
+
+    def generate_shares(self, plain_text, number_of_shares=3):
```

Comment:
I recommend taking example on https://github.com/OpenMined/PySyft/blob/4518a44700cefcd6efd4b43dbf7bf8ae79c52afb/syft/frameworks/torch/tensors/interpreters/additive_shared.py#L334
i.e. remove the noise you've added to the next share instead of removing the sum of all on the last share

Reply:
What is the difference? in other words, why would you choose this over that?

Reply:
ping @LaRiffle

Reply:
I did, I'm looking at how crypten handles it